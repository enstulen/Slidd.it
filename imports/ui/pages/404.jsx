import React, { Component } from 'react';
import { mount } from 'react-mounter';
import NavbarHeader from '../NavbarHeader';
import Welcome from '../../ui/pages/welcome.jsx';


class FourOFour extends Component {
  constructor(props) {
    super(props);

    this.handleButtonPress = this.handleButtonPress.bind(this);
  }

  handleButtonPress(event) {
    event.preventDefault();
    window.location = 'http://www.slidd.it';
  }

  render() {
    return (
      <div>
        <div className="navbar">
          <NavbarHeader />
        </div>
        <center><div id="centerBox">
          <h1><big><big>404</big></big></h1>
          <h2>Page not found</h2>
          <button type="button" className="btn btn-primary btn-lg" onClick={this.handleButtonPress}>Go to main page</button>
        </div></center>
      </div>
    );
  }
}

export default FourOFour;
