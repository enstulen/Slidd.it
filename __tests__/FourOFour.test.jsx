import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import { FourOFour } from '../imports/ui/pages/404.jsx';

describe('404 component', () => {
  it('Renders correctly', () => {
    const rendered = renderer.create(
      <FourOFour lectures={[]}/>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('Correct length', () => {

    const wrapper = shallow(<FourOFour />);
    expect(wrapper.find('.navbar').length).toEqual(1);
    expect(wrapper.find('.btn-lg').length).toEqual(1);

  });
});
