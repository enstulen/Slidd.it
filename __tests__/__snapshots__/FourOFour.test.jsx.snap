// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 component Renders correctly 1`] = `
<div>
  <div
    className="navbar"
  >
    <div>
      <nav
        className="navbar navbar-default"
      >
        <div
          className="container-fluid"
        >
          <div
            className="navbar-header"
          >
            <button
              aria-expanded="false"
              className="navbar-toggle collapsed"
              data-target="#bs-example-navbar-collapse-1"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="sr-only"
              >
                Toggle navigation
              </span>
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
            </button>
            <button
              className="invisibleButton"
              onClick={[Function]}
            >
              <img
                alt="Slidd.it"
                className="img-responsive"
                src="/logo3.png"
                width="180"
              />
            </button>
          </div>
          <div
            className="collapse navbar-collapse"
            id="bs-example-navbar-collapse-1"
          >
            <ul
              className="nav navbar-nav"
            >
              <li>
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                className="dropdown"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  className="dropdown-toggle"
                  data-toggle="dropdown"
                  role="button"
                >
                  Active lectures 
                  <span
                    className="caret"
                  />
                </a>
                <ul
                  aria-labelledby="dropdownMenu1"
                  className="dropdown-menu"
                >
                  <li>
                    <button
                      className="liButton"
                      onClick={[Function]}
                    >
                      Example class
                    </button>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </div>
  <center>
    <div
      id="centerBox"
    >
      <img
        alt="404Chicken"
        src="/Hoonegif.gif"
        width="300"
      />
      <h1>
        <big>
          <big>
            404
          </big>
        </big>
      </h1>
      <h2>
        Page not found
      </h2>
      <button
        className="btn btn-primary btn-lg"
        onClick={[Function]}
        type="button"
      >
        Go to main page
      </button>
    </div>
  </center>
</div>
`;
